.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,86 @@
+## Type of change
+
+<!-- Mandatory
+Please delete options that are not relevant
+-->
+
+- [ ] New feature
+- [ ] Bug fix
+- [ ] Enhancement
+- [ ] Breaking change
+
+## What does this PR do?
+
+<!-- Mandatory
+Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
+-->
+
+## Why is it important?
+
+<!-- Mandatory
+Explain here the WHY, or the rationale/motivation for the changes.
+-->
+
+## Checklist
+
+<!-- Mandatory
+Add a checklist of things that are required to be reviewed in order to have the PR approved
+
+List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply.
+-->
+
+- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing **unit** tests pass locally with my changes
+- [ ] New and existing **integration** tests pass locally with my changes
```

Comment:
CI should answer these questions

Reply:
Well, we's expect the sender to run the tests locally first, right? At least the unit tests

Reply:
Once the PR is created it will be run by CI anyways. I think it is good practice to run unit tests before pushing the PR, but in the end CI will tell us the truth.

I like the idea of the check list, so to not forget some common tasks (e.g. have you updated docs/configs). But I would prefer have the check list small. Things that are automatically checked anyways don't need to be in the list.

Reply:
Agreed in the shortness of the checklist is a must. We'll remove what is not necessary.

About the unit test part, I'd suggest keeping it, or at least point to the place where it explains how to run them with a nice experience.